scala-generator/src/main/scala/models/Play2Json.scala

```diff
@@ -144,19 +144,39 @@ case class Play2Json(
   }
 
   private[this] def readersWithDiscriminator(union: ScalaUnion, discriminator: String): String = {
+    val defaultDiscriminatorTypeName: Option[String] = unionTypesWithNames(union).flatMap { case (t, typeName) =>
```

Comment:
if i am reading this properly, this says that if default isn't defined, then nothing is? if a there is no default union type, should we just pick the first?